src/circuits/mimc/mimc_hash.hpp

```diff
@@ -0,0 +1,42 @@
+// DISCLAIMER:
+// Content taken and adapted from:
+// https://github.com/HarryR/ethsnarks/blob/master/src/gadgets/mimc.hpp
+
+#ifndef __ZETH_MIMC_HASH_HPP__
+#define __ZETH_MIMC_HASH_HPP__
+
+#include "snarks_alias.hpp"
+#include "mimc.hpp"
+
+namespace libzeth {
+
+template<typename FieldT>
+class MiMC_hash_gadget:public libsnark::gadget<FieldT> {
```

Comment:
typedefs `hash_value_type;` and  `merkle_authentication_path_type;` are not specified for this hash. Calling `hashT::hash_value_type` will fail (used in the `merkle_tree` implementation of libsnark to define the type of the hashes)

Reply:
See: https://github.com/scipr-lab/libsnark/blob/master/libsnark/common/data_structures/merkle_tree.hpp#L42-L43